aiidalab_widgets_base/viewers.py

```diff
@@ -1165,13 +1165,15 @@ def __init__(self, bands, **kwargs):
 
 
 @register_viewer_widget("process.workflow.workchain.WorkChainNode.")
-class WorkChainNodeViewerWidget(ipw.HTML):
-    def __init__(self, workchain, **kwargs):
-        self.workchain = workchain
-
-        # Displaying reports only from the selected workchain,
-        # NOT from its descendants
-        report = get_workchain_report(self.workchain, "REPORT", max_depth=1)
+@register_viewer_widget("process.calculation.calcfunction.CalcFunctionNode.")
+@register_viewer_widget("process.workflow.workfunction.WorkFunctionNode.")
```

Comment:
I did not want to do that since in my app and QeApp we already have a viewer for CalcJobNodes, which dynamically fetches the output of the calculations.

https://github.com/aiidalab/aiidalab-qe/blob/b827240c5609cb6ead54bebc972d1c78a4d26181/aiidalab_qe/widgets.py#L345

Now that I looked into it, QeApp also has its own viewer for WorkChainNode

https://github.com/aiidalab/aiidalab-qe/blob/b827240c5609cb6ead54bebc972d1c78a4d26181/aiidalab_qe/node_view.py#L411

I am not sure what happens when two different viewers are registered for the same node type, since then it depends on the order of when the viewer is registered I think. Let me know if you want me to add it anyway and test it.

Reply:
> I am not sure what happens when two different viewers are registered for the same node type, since then it depends on the order of when the viewer is registered I think. Let me know if you want me to add it anyway and test it.

TBH, this is not 100% clear to me either.

I guess what we really want is some default viewers defined in AWB, but also the ability to overwrite them if needed for a specific app. But I am not sure the current setup offers a clear path to achieve that.

Would you have some time/energy to investigate this question further? Feels to me like an important point to clarify before we move forward.

Reply:
Unfortunately I need to switch working on a different project this and next week so don't have time to dig deeper right now. But it's okay for me to keep this PR open till then. The important PRs for me have been merged so getting a release out might be nice (there's been a couple since last release I think).

Reply:
Ok, then @danielhollas you can go ahead and add all four process types. It looks like the viewers that are defined for specific apps will override those "default" ones.

Reply:
Done, thanks.